backends/dpdk/dpdkContext.h

```diff
@@ -53,6 +52,128 @@ struct TableAttributes {
     std::vector<std::pair<cstring, cstring>> tableKeys;
 };
 
+enum TableTypes {
+    MATCH = 0
+};
+
+enum MatchType {
+    exact = 0,
+    ternary = 1,
+    lpm = 2,
+    range = 3,
+    selector = 4
+};
+
+struct MatchKeyFieldValue {
+    cstring fieldName;
+    cstring value;
+    bool isValue;
+    cstring mask;
+    bool isMask;
+    cstring rangeStart;
+    cstring rangeEnd;
+    bool isRange;
+    unsigned int prefixLength;
+    bool isPrefixLength;
+
+    MatchKeyFieldValue() {
+        fieldName = "";
+        value = "";
+        isValue = false;
+        mask = "";
+        isMask = false;
+        rangeStart = "";
+        rangeEnd = "";
+        isRange = false;
+        prefixLength = 0;
+        isPrefixLength = false;
+    }
+};
+
+struct MatchKeyField {
+    cstring name;
+    bool isInstanceName;
+    cstring instanceName;
+    bool isFieldName;
+    cstring fieldName;
+    MatchType matchType;
+    unsigned int startBit;
+    unsigned int bitWidth;
+    unsigned int bitWidthFull;
+    unsigned int index;
+    unsigned int position;
+    MatchKeyField() {
+        name = "";
+        instanceName = "";
+        isInstanceName = false;
+        fieldName = "";
+        isFieldName = false;
+        matchType = exact;
+        startBit = 0;
+        bitWidth = 0;
+        bitWidthFull = 0;
+        index = 0;
+        position = 0;
+    }
+};
+
+struct ImmediateFields {
+    cstring param_name;
+    int param_handle;
+    int dest_start;
+    int dest_width;
+    ImmediateFields() {
+        param_name = "";
+        param_handle = -1;
+        dest_start = -1;
+        dest_width = 0;
+    }
+};
+
+struct LookupHwBlocks {
+    cstring resource;
+    int resource_id;
+    std::vector<ImmediateFields*> immediate_fields;
+        LookupHwBlocks() {
+            resource = "";
+            resource_id = 0;
+        }
+};
+
+struct LookupMatchAttributes {
+    std::vector<LookupHwBlocks*> hardware_blocks;
+};
+
+
+/* This structure hold match value lookup table attributes 
+ * collected from declarations */
+struct P4MatchLookupTableInfo {
+    TableTypes tableType;
+    unsigned int handle;
+    cstring name;
+    cstring ctrlName;
```

Comment:
Control plane name/external name needs to be added here. We need to emit two names for each table/extern. One name("target_name") matching with the name in the spec file and the other "name" same as ControlPlaneName().
For example, https://wiki.ith.intel.com/pages/viewpage.action?pageId=2253848776#ChangeRequests-Change_Request_CR00021